tee-worker/service/src/parentchain_handler.rs

```diff
@@ -142,6 +146,49 @@ where
 				return Ok(until_synced_header)
 			}
 
+			let mut events: Vec<Events> = vec![];
+			for block in &block_chunk_to_sync {
+				let block_events = self.parentchain_api.events(Some(block.block.hash()))?;
+				events.push(block_events);
+			}
+			block_chunk_to_sync
+				.iter_mut()
+				.enumerate()
+				.for_each(|(block_index, signed_block)| {
+					let mut extrinsics: Vec<OpaqueExtrinsic> = vec![];
+					let block_events = events.get(block_index).unwrap();
+					for (i, xt) in signed_block.block.extrinsics.iter().enumerate() {
+						// Check if the tx was successful
+						let success = block_events
+							.iter()
+							.filter(|event| {
+								if let Ok(event) = event {
+									event.pallet_name() == "System"
+										&& event.variant_name() == "ExtrinsicSuccess"
+										&& event.phase() == Phase::ApplyExtrinsic(i as u32)
+								} else {
+									false
+								}
+							})
+							.count() > 0;
+						if !success {
+							warn!(
+								"block:{:?}, extrinsic index: {:?}, success: {:?}",
+								&signed_block.block.header.number, i, success,
+							);
+						}
+
+						// `indirect_calls_executor` should know the if tx was successful.
+						// if we change the block type `sp_runtime::generic::Block` or `sp_runtime::generic::SignedBlock`,
+						// this change will affect many structs or files, like block_importer, block_import_dispatcher, consensus and so on.
+						// `OpaqueExtrinsic` (Vec<u8>) encoded value contains tx status, and decode Vec<u8> in `indirect_calls_executor`,
+						// this solution is probably the least change
```

Comment:
Maybe let's move this comment to line 149 for better readability, it also helps to explain why we try to get the events of parentchain blocks